tests/integration/targets/vmware_dvs_host/tasks/main.yml

```
@@ -0,0 +1,67 @@
+---
+- import_role:
+    name: prepare_vmware_tests
+  vars:
+    setup_datacenter: true
```

```suggestion
    setup_attach_host: true
```

The CI pipeline is still a black box to me (not as black as it used to be, but only marginally) but could you try to `setup_attach_host` instead of `setup_datacenter`?

And then don't loop but just use `esxi_hostname: '{{ esxi1 }}'`? I won't promise anything but maybe this will work.

Btw, I'm pretty sure you don't need the uplink / distributed portgroup tasks for this test. And I also think you don't have to detach the host before you delete switch.